internal/dcache/file_manager/file_manager.go

```diff
@@ -65,35 +65,102 @@ type fileIOManager struct {
 
 var fileIOMgr fileIOManager
 
-func NewFileIOManager(workers int, numReadAheadChunks int, numStagingChunks int, bufSize int, maxBuffers int) {
-	common.Assert(workers > 0)
-	common.Assert(numReadAheadChunks > 0)
-	common.Assert(numStagingChunks > 0)
+func NewFileIOManager() error {
+	//
+	// A worker runs either readChunk() or writeChunk(), so this is the number of chunks we can be
+	// reading/writing in parallel. fileIOManager is one for the entire blobfuse process so these
+	// chunks can be spread across multiple files and served from multiple nodes/RVs.
+	// Note that fir writeChunk one worker is used up regardless of the NumReplicas setting. Each
+	// replica write uses us one ReplicationManager worker but only one fileIOManager worker per
+	// MV write (not replica write).
+	// Since all these reads/writes may be served from different nodes/RVs, the limiting factor would
+	// be the n/w b/w of this node. We need enough parallel readChunk/writeChunk for maxing out the
+	// n/w b/w of the node. Keeping small files in mind, and given that go routines are not very
+	// expensive, we keep 1000 workers.
+	//
+	workers := 1000
```

Comment:
We can define a constant for this and other variables instead of hardcoding the values.